test/ech_test.c

```diff
@@ -34,6 +562,310 @@ const OPTIONS *test_get_options(void)
     return test_options;
 }
 
+/*
+ * For the relevant test vector in our array above:
+ * - try decode
+ * - if not expected to decode, we're done
+ * - check we got the right number of keys/ECHConfig values
+ * - do some calls with getting info, downselecting etc. and
+ *   check results as expected
+ * - do a write_pem call on the results
+ * - flush keys 'till now and check they're all gone
+ */
+static int ech_ingest_test(int run)
+{
+    OSSL_ECHSTORE *es = NULL;
+    OSSL_ECH_INFO *ei = NULL;
+    BIO *in = NULL, *out = NULL;
+    int i, rv = 0, keysb4, keysaftr, entsb4, entsaftr;
+    ingest_tv_t *tv = &ingest_tvs[run];
+    time_t now = 0;
+
+    if ((in = BIO_new(BIO_s_mem())) == NULL
+        || BIO_write(in, tv->tv, tv->len) <= 0
+        || (out = BIO_new(BIO_s_mem())) == NULL
+        || (es = OSSL_ECHSTORE_new(NULL, NULL)) == NULL)
+        goto end;
+    if (verbose)
+        TEST_info("Iteration: %d %s", run + 1, tv->name);
+    /* just in case of bad edits to table */
+    if (tv->pemenc != 1 && tv->pemenc != 0) {
+        TEST_info("Bad test vector entry");
+        goto end;
+    }
+    if (tv->pemenc == 1
+        && !TEST_int_eq(OSSL_ECHSTORE_read_pem(es, in, OSSL_ECH_NO_RETRY),
+                        tv->read)) {
+        TEST_info("OSSL_ECSTORE_read_pem unexpected fail");
```

Comment:
This error message is slightly confusing, because in some tests we are expecting to fail - so we hit this on an unexpected success!

Reply:
did s/unexpected fail/unexpected result/